app/routes/task_routes.py

```diff
@@ -1 +1,141 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import abort, Blueprint, jsonify, make_response, request, Response
+from app.models.task import Task
+from app.models.goal import Goal
+from ..db import db
+from sqlalchemy import cast, String, Integer
+from .route_utilities import validate_model
+from datetime import datetime, timezone
+import os
+import requests
+
+bp = Blueprint("tasks_bp", __name__, url_prefix = "/tasks")
+
+@bp.post("")
+def create_task():
+    request_body = request.get_json()
+    try:
+        new_task = Task.from_dict(request_body)
+
+    except KeyError as error:
+        response = {"details": "Invalid data"}
+        abort(make_response(response, 400))
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return {"task": new_task.to_dict()}, 201
+
+@bp.get("")
+def get_all_tasks():
+    query = db.select(Task)
+    tasks = db.session.execute(query).scalars().all()
+
+    sort_order_param = request.args.get("sort", "asc")
+    if sort_order_param == "desc":
+        query = query.order_by(Task.title.desc())
+    else:
+        query = query.order_by(Task.title.asc())
+
+    title_param = request.args.get("title")
+    if title_param:
+        query = query.where(Task.title.ilike(f"%{title_param}%"))
+    
+    description_param = request.args.get("description")
+    if description_param:
+        query = query.where(Task.description.ilike(f"%{description_param}%"))
+
+    completed_at_param =request.args.get("completed_at")
+    if completed_at_param:
+        query = query.where(Task.completed_at.ilike(f"%{completed_at_param}%"))
+
+    is_complete_param = request.args.get("is_complete")
+    if is_complete_param:
+        # is_complete_bool = is_complete_param.lower() == "true"
+        # query = query.where(Task.is_complete == is_complete_bool)
+
+        if is_complete_param.lower() == "true":
+            query = query.where(Task.task_completed == True)
+        elif is_complete_param.lower() == "false":
+            query = query.where(Task.task_completed == False)
+
+    query = query.order_by(Task.id)
+    tasks = db.session.scalars(query)
```

Comment:
I would say that this a great opportunity for a refactor. Note how it takes up most of the function logic. We lose sight of the function's purpose, handling the a request to get all tasks. Right now, it seems like its purpose is to build a query based off of certain filters.